src/status_im2/contexts/chat/messages/list/view.cljs

```diff
@@ -72,16 +72,20 @@
   "compute data used to render message list, including pinned message list and message list in chats"
   [{:keys [group-chat chat-id public? community-id admins space-keeper show-input? edit-enabled
            in-pinned-view?]}]
-  (let [current-public-key                                       (rf/sub [:multiaccount/public-key])
-        {:keys [can-delete-message-for-everyone?] :as community} (rf/sub [:communities/community
-                                                                          community-id])
-        group-admin?                                             (get admins current-public-key)
-        community-admin?                                         (when community (community :admin))
-        message-pin-enabled                                      (and (not public?)
-                                                                      (or (not group-chat)
-                                                                          (and group-chat
-                                                                               (or group-admin?
-                                                                                   community-admin?))))]
+  (let [current-public-key                        (rf/sub [:multiaccount/public-key])
+        {:keys [can-delete-message-for-everyone?
+                admin-settings]
+         :as   community}                         (rf/sub [:communities/community
+                                                           community-id])
+        {:keys [pin-message-all-members-enabled?]} admin-settings
+        group-admin?                              (get admins current-public-key)
+        community-admin?                          (when community (community :admin))
+        message-pin-enabled                       (and (not public?)
+                                                       (or (not group-chat)
+                                                           (and group-chat
+                                                                (or group-admin?
+                                                                    pin-message-all-members-enabled?
+                                                                    community-admin?))))]
     {:group-chat                       group-chat
```

Comment:
I have the impression that this entire `get-render-data` function smells and it should be a subscription and properly unit tested. It's heavy on logic, doesn't render anything and it uses a bunch of other subscriptions, which is a clear use case for layer-3 subs.

Reply:
Agree, after a bit dig, this is much larger scope than I expected, affects a lot in the chat view.
I'll do some research and see what can be done. And, of course, in a separate issue.

Reply:
You're right, too much for this PR. Thanks for digging into this problem :100: